### PR TITLE
refactor(core/link-to-dfn) split into internal and external linking modules

### PR DESCRIPTION
**Pull request by sidvishnoi**
Why? In upcoming xref feature, we only need to find references to those `<a>` which have no `dfn` locally. So splitting it into internal linking (runs first) and external linking (to run after core/xref and core/data-cite) modules.

TODO:
- [x] Split
- [ ] DRY
- [ ] Remove jQuery

### CODE REVIEW

**Comment by sidvishnoi** [comment]
Regarding git history:
I've renamed `link-to-dfn` to `link-to-dfn-external` so the file history remains. 
If I were to edit `link-to-dfn-external` now and we squash-merge the edits & rename, will the git history of this file be fine? Maybe we should merge it in two parts to keep history, otherwise it's all fine.

File: js/profile-w3c-common.js [hunk 1]
```diff
@@ -49,10 +49,11 @@ define(
     "core/best-practices",
     "core/figures",
     "core/webidl",
+    "core/link-to-dfn-internal",
```
Comment (marcoscaceres) [inline js/profile-w3c-common.js]: Unfortunately, I don’t think we can’t rename this, as it would break other profiles.

Comment (sidvishnoi) [reply js/profile-w3c-common.js]: Alternatives? We do need to separate out linking of internal and external definitions, as I believe.

Comment (marcoscaceres) [reply js/profile-w3c-common.js]: doing the separation is fine, but this module should remain as is (and retain backwards compatibility). The new external xrefs can just live in “*-external.js”

Comment (sidvishnoi) [reply js/profile-w3c-common.js]: > doing the separation is fine, but this module should remain as is

Didn't quite get that. "remain same" in what sense?

I actually want to separate it into two parts as:
- I do not want `link-to-dfn` to emit `"warn", Linking error: not matching <dfn>` while it has just looked for internal dfns

Comment (marcoscaceres) [reply js/profile-w3c-common.js]: That’s ok, but that shouldn’t require the module to be renamed. The external xref module could be imported by this module, for instance. It doesn’t necessarily also need to be a separate plugin.

Comment (marcoscaceres) [reply js/profile-w3c-common.js]: > Didn't quite get that. "remain same" in what sense?

To be clear, in the sense that the external reference code should be a progressive enhancement, and not a breaking change.

Comment (marcoscaceres) [reply js/profile-w3c-common.js]: I’ll try to be online in about 2.5 hours so we can discuss... still 😴 🛏  here (yay jet lag!).

**Comment by sidvishnoi** [comment]
Closing this in favour of new discussed strategy

**Pull request closed without merging**